project/app/views.py

```diff
@@ -124,54 +92,58 @@ class UpdateProfileView(APIView):
     permission_classes = [IsAuthenticated, RequiredAdmin]
 
     def put(self, request, pk):
-        user_profile = UserProfile.objects.get(user_id=pk)
-        profile_serializer = UserProfileSerializer(instance=user_profile, data=request.data)
+        try:
+            user_profile = UserProfile.objects.get(user_id=pk)
+            profile_serializer = UserProfileSerializer(instance=user_profile, data=request.data)
 
-        if not profile_serializer.is_valid():
-            return Response(status=status.HTTP_404_NOT_FOUND)
-        profile_serializer.save()
-        return Response(profile_serializer.data)
+            if not profile_serializer.is_valid():
+                return Response({"msg: data you entered is wrong"}, status=status.HTTP_404_NOT_FOUND)
+            profile_serializer.save()
+        except:
+            return Response({
+                "msg": "user with this id does not exists"
+            })
 
 
-class AdminAddUserView(APIView):
-    permission_classes = [IsAuthenticated, RequiredAdmin, AdminCanCreate]
+class AddUserView(APIView):
+    permission_classes = [IsAuthenticated, RequiredAdmin | RequiredManager]
 
     def post(self, request):
         serializer = UserSerializer(data=request.data)
         if not serializer.is_valid():
-            return Response(status=status.HTTP_404_NOT_FOUND)
+            return Response(serializer.errors, status=status.HTTP_404_NOT_FOUND)
         serializer.save()
-        email = request.POST.get("email")
-        user = CustomUser.objects.get(email=email)
-        user.is_active = True
-        user.save()
-        return Response(serializer.data)
+        try:
+            email = request.POST.get("email")
```

Comment:
It should be done within the serializer save method or we can update the value using the instance we are getting from the serializer.save().
However, the best way will be to do it in the serializer save method itself.